freqtrade/optimize/hyperopt.py

```diff
@@ -578,20 +707,332 @@ def _format_results_explanation_string(self, results_metrics: Dict) -> str:
+        else:
+            void = True
+        # send back the updated optimizer only in non shared mode
+        if not is_shared:
+            opt = Hyperopt.opt_clear(opt)
+            # is not a replica in shared mode
+            optimizers.put(opt)
+        # NOTE: some results at the beginning won't be published
+        # because they are removed by filter_void_losses
+        rs = opt.rs
+        if not void:
+            # the tuple keys are used to avoid computation of done points by any optimizer
+            results_shared.update({tuple(Hyperopt.params_Xi(v)): (v["loss"], jobs - 1)
+                                   for v in void_filtered})
+            # in multi opt mode (non shared) also track results for each optimizer (using rs as ID)
+            # this keys should be cleared after each batch
+            Xi, yi = results_shared[rs]
+            Xi = Xi + tuple((Hyperopt.params_Xi(v)) for v in void_filtered)
+            yi = yi + tuple(v["loss"] for v in void_filtered)
+            results_shared[rs] = (Xi, yi)
+            # this is the counter used by the optimizer internally to track the initial
+            # points evaluated so far..
+            initial_points = opt._n_initial_points
+            # set initial point flag and optimizer random state
+            for n, v in enumerate(void_filtered):
+                v['is_initial_point'] = initial_points - n > 0
+                v['random_state'] = rs
+            results_batch.put(void_filtered)
+
+    def parallel_opt_objective(self, n: int, optimizers: Queue, jobs: int,
+                               results_shared: Dict, results_batch: Queue):
+        """
+        objective run in multi opt mode, optimizers share the results as soon as they are completed
+        """
+        self.log_results_immediate(n)
+        is_shared = self.shared
+        opt = self.opt_state(is_shared, optimizers)
+        sss = self.search_space_size
+        asked: Dict[Tuple, Any] = {tuple([]): None}
+        asked_d: Dict[Tuple, Any] = {}
+
+        # fit a model with the known points, (the optimizer has no points here since
+        # it was just fetched from the queue)
+        rs = opt.rs
+        Xi, yi = self.Xi[rs], self.yi[rs]
+        # add the points discovered within this batch
+        bXi, byi = results_shared[rs]
+        Xi.extend(list(bXi))
+        yi.extend(list(byi))
+        if Xi:
+            opt.tell(Xi, yi)
+        told = 0  # told
+        Xi_d = []  # done
+        yi_d = []
+        Xi_t = []  # to do
+        # if opt.void == -1 the optimizer failed to give a new point (between dispatches), stop
+        # if asked == asked_d  the points returned are the same, stop
+        # if opt.Xi > sss the optimizer has more points than the estimated search space size, stop
+        while opt.void != -1 and asked != asked_d and len(opt.Xi) < sss:
+            asked_d = asked
+            asked = opt.ask(n_points=self.ask_points, strategy=self.lie_strat())
```

Comment:
Note to self:
providing strategy to `opt.ask` does not make sense in this way - it'll decide if the optimize function should be minimized (`cl_min`), optimized (`cl_max`), or averaged (`cl_mean`).

Using a random strategy for this (or letting the user pick) does not make sense.
What could make sense is to let the hyperopt-loss function determine if better is higher, or if better is lower (optimize vs. minimize). 
This could remove some of "negativation" of loss-functions like Sharpe and Sortino, where originally higher is better.